docs/ibis-for-dplyr-users.ipynb

```diff
@@ -0,0 +1,774 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Ibis for dplyr Users"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "[R](https://www.r-project.org/) users familiar with [dplyr](https://dplyr.tidyverse.org/) and other packages in the [Tidyverse](https://www.tidyverse.org/) are likely find Ibis familiar.\n",
+    "In fact, some Ibis features were even inspired by similar features in the [Tidyverse](https://www.tidyverse.org/).\n",
+    "\n",
+    "However, due to differences between Python and R and the design and goals of Ibis itself, you may notice some big differences right away:\n",
+    "\n",
+    "- **No pipe:** The handy [magrittr pipe](https://magrittr.tidyverse.org/) (`%>%`) or R's newer native pipe (`|>`) don't exist in Python so you instead have to chain sequences of operations together with a period (`.`).\n",
+    "- **No unquoted column names:** Non-standard evaluation is common in R but not present in Python. To reference a column in Ibis, you can pass a string, property on a table (e.g., `tbl.some_column`), or you can make use of [selectors](https://ibis-project.org/api/selectors/).\n",
+    "- **Ibis is lazy by default:** Similar to [dbplyr](https://dbplyr.tidyverse.org/) and its `collect()` method, Ibis does not evaluate our queries until we call `.execute()`. For the purposes of this document, we set `ibis.options.interactive = True` which calls `execute` automatically for us and prints a nicely-formatted table."
```

Comment:
When `ibis.options.interactive = True`, Ibis also adds a `LIMIT 11` to the query. That way the backend engine is only processing as much data is necessary to display 10 rows of the result in the nicely-formatted table (plus an ellipsis if there are more than 10 rows in the result... that's why it's 11 not 10). IIRC this is the same thing dbplyr does when you print a `tbl_sql`.

Reply:
Ah, the 11 vs 10 thing is a nice trick. Accepted basically what you wrote. Thanks.

Reply:
It would be good to clarify that even when `ibis.options.interactive = True`, calling `execute()` returns a pandas DataFrame with _all_ the result rows, not just the first 10.

Maybe add an example showing how you can store the result as a Python variable which is a pandas DataFrame?

Reply:
I think that makes sense. I added some text and an example in 0609126f44e347461919e3a216c874abd86fd879.